### PR TITLE
CLOUD-656 - Add support for chaos-mesh 2.0.4 and kubernetes 1.22

### PR DESCRIPTION
**Pull request by tplavcic**
[CLOUD-656](https://jira.percona.com/browse/CLOUD-656) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

File: e2e-tests/operator-self-healing-chaos/run [hunk 1]
```diff
@@ -115,7 +113,9 @@ main() {
 	deploy_chaos_mesh $test_namespace
 
 	desc 'create PXC cluster'
-	spinup_pxc "$cluster" "$test_dir/conf/$cluster.yml"
+	cp -f "$test_dir/conf/$cluster.yml" "$tmp_dir/$cluster.yml"
+	yq w -i "$tmp_dir/$cluster.yml" "spec.initImage" "$IMAGE"
+	spinup_pxc "$cluster" "$tmp_dir/$cluster.yml"
```
Comment (delgod) [inline e2e-tests/operator-self-healing-chaos/run:118]: I believe it is better to change `functions`
```
--- e2e-tests/functions
+++ e2e-tests/functions
@@ -684,6 +684,7 @@ cat_config() {
                | $sed -e "s#image:.*-pxc\([0-9]*.[0-9]*\)\{0,1\}\$#image: $IMAGE_PXC#" \
                | $sed -e "s#image:.*\/percona-xtradb-cluster:.*\$#image: $IMAGE_PXC#" \
                | $sed -e "s#image:.*-pmm\$#image: $IMAGE_PMM#" \
+               | $sed -e "s#image:.*-init\$#image: $IMAGE#" \
                | $sed -e "s#image:.*-backup\$#image: $IMAGE_BACKUP#" \
                | $sed -e "s#image:.*-proxysql\$#image: $IMAGE_PROXY#" \
                | $sed -e "s#image:.*-haproxy\$#image: $IMAGE_HAPROXY#" \
```

**Review by delgod — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| storage | passed |
| scheduled-backup | passed |
| scaling | passed |
| haproxy | passed |
| big-data | passed |
| recreate | passed |
| self-healing | passed |
| self-healing-chaos | failed |
| scaling-proxysql | passed |
| init-deploy | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| limits | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | failed |
| operator-self-healing | passed |
| operator-self-healing-chaos | failed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| pitr | passed |
| proxysql-sidecar-res-limits | passed |
| smart-update | passed |
| users | passed |
| tls-issue-self | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| upgrade-consistency | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1066/commits/1548adc48b833f693a3914c7186a71abfd648bbd
image: `perconalab/percona-xtradb-cluster-operator:PR-1066-1548adc4`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scheduled-backup | passed |
| haproxy | passed |
| recreate | passed |
| upgrade-haproxy | passed |
| big-data | failed |
| scaling | passed |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | failed |
| restore-to-encrypted-cluster | passed |
| scaling-proxysql | passed |
| upgrade-proxysql | passed |
| init-deploy | passed |
| security-context | passed |
| limits | passed |
| demand-backup | passed |
| smart-update | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| pitr | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| upgrade-consistency | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | failed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-cert-manager-ref | failed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1066/commits/a3e3755e715c777b0f54c4577acb80290ee9fc55
image: `perconalab/percona-xtradb-cluster-operator:PR-1066-a3e3755e`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| haproxy | passed |
| scaling | passed |
| upgrade-haproxy | passed |
| scheduled-backup | passed |
| big-data | passed |
| recreate | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| limits | passed |
| monitoring-2-0 | passed |
| smart-update | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| affinity | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| auto-tuning | passed |
| pitr | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1066/commits/801a2c2faa61b55220a33254337030c2c0205566
image: `perconalab/percona-xtradb-cluster-operator:PR-1066-801a2c2f`

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| big-data | passed |
| upgrade-haproxy | passed |
| scheduled-backup | passed |
| haproxy | passed |
| self-healing | passed |
| init-deploy | passed |
| self-healing-chaos | passed |
| limits | passed |
| self-healing-advanced | passed |
| recreate | passed |
| monitoring-2-0 | passed |
| self-healing-advanced-chaos | passed |
| restore-to-encrypted-cluster | passed |
| affinity | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| one-pod | passed |
| operator-self-healing-chaos | passed |
| pitr | passed |
| auto-tuning | passed |
| demand-backup-encrypted-with-tls | failed |
| proxysql-sidecar-res-limits | passed |
| scaling | passed |
| users | passed |
| scaling-proxysql | passed |
| tls-issue-self | passed |
| security-context | passed |
| tls-issue-cert-manager | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1066/commits/801a2c2faa61b55220a33254337030c2c0205566
image: `perconalab/percona-xtradb-cluster-operator:PR-1066-801a2c2f`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| security-context | passed |
| big-data | passed |
| upgrade-haproxy | passed |
| haproxy | passed |
| recreate | passed |
| storage | passed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| self-healing | passed |
| limits | passed |
| demand-backup | passed |
| self-healing-chaos | passed |
| monitoring-2-0 | passed |
| pitr | passed |
| self-healing-advanced | passed |
| affinity | passed |
| demand-backup-encrypted-with-tls | passed |
| scheduled-backup | passed |
| self-healing-advanced-chaos | passed |
| one-pod | passed |
| operator-self-healing | passed |
| auto-tuning | passed |
| operator-self-healing-chaos | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1066/commits/801a2c2faa61b55220a33254337030c2c0205566
image: `perconalab/percona-xtradb-cluster-operator:PR-1066-801a2c2f`

**Review by hors — APPROVED** [review]

**Pull request merged**